src/backend/executor/execQual.c

```diff
@@ -2256,10 +2256,6 @@ ExecMakeTableFunctionResult(ExprState *funcexpr,
 
 	callerContext = CurrentMemoryContext;
 
-	funcrettype = exprType((Node *) funcexpr->expr);
```

Comment:
I think you should have just replaced the condition `funcexpr &&` with an assert before the dereference here.

Reply:
Done